notebooks/curation_api/python/src/dataset.py

```diff
@@ -1,18 +1,103 @@
+import json
+
 import boto3
 import logging
 import os
+import re
 import requests
 import threading
 from botocore.credentials import RefreshableCredentials
 from botocore.session import get_session
 from datetime import datetime, timezone
+from typing import Tuple
 
 from src.utils.logger import get_custom_logger
 from src.utils.http import url_builder, get_headers
 
 
 logger = get_custom_logger()
 
+UUID_REGEX = r"[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}"
+DATASET_ID_REGEX = f"(?P<dataset_uuid>{UUID_REGEX})"
+CURATOR_TAG_PREFIX_REGEX = r"(?P<tag_prefix>.*)"
+EXTENSION_REGEX = r"(?P<extension>h5ad)"
+
+
+def get_identifier_type_and_value(identifier: str) -> Tuple[str, str]:
+    identifier_type = None
+    identifier_value = identifier
+    if re.match(f"^{DATASET_ID_REGEX}$", identifier):
+        # identifier is a uuid
+        identifier_type = "dataset_uuid"
+    else:
+        # CURATOR_TAG_PREFIX_REGEX is superfluous; leaving in to match lambda handler code; may use later
+        matched = re.match(f"({DATASET_ID_REGEX}|{CURATOR_TAG_PREFIX_REGEX})\\.{EXTENSION_REGEX}$", identifier)
+        if matched:
+            matches = matched.groupdict()
+            if dataset_uuid := matches.get("dataset_uuid"):
+                identifier_type = "dataset_uuid"
+                identifier_value = dataset_uuid
+            else:
+                identifier_type = "curator_tag"
+
+    if not identifier_type:
+        raise Exception(f"The identifier '{identifier}' must either 1) include a '.h5ad' suffix OR 2) be a uuid")
+
+    return identifier_value, identifier_type
```

Comment:
super nit: I would prefer returning (type, value) instead to match the function name order